gcc/rust/gource-gccrs.sh

```diff
@@ -0,0 +1,108 @@
+#!/usr/bin/env bash
+set -e
+
+RESOLUTION="1600x1080"
+output=""
+while getopts o:r: flag
+do
+    case "${flag}" in
+        o) output=${OPTARG};;
+        r) RESOLUTION=${OPTARG};;
+    esac
+done
+
+# we require gource and ffmpeg to generate this stuff
+if ! [ -x "$(command -v git)" ]; then
+  echo 'Error: git is not installed.' >&2
+  exit 1
+fi
+if ! [ -x "$(command -v gource)" ]; then
+  echo 'Error: gource is not installed.' >&2
+  exit 1
+fi
+if ! [ -x "$(command -v ffmpeg)" ]; then
+  echo 'Error: ffmpeg is not installed.' >&2
+  exit 1
+fi
+
+# are we in the root of the repo?
+if [ ! -d ".git" ]; then
+    echo ".git directory not found, this script must be run at root repo"
+    exit 1
+fi
+
+# if the output already exists then we should stop
+if test -f "$output"; then
+    echo "$output already exists."
+    exit 1
+fi
+
+# is cargo-gccrs here?
+if [ ! -d "cargo-gccrs" ]; then
+    git clone git@github.com:Rust-GCC/cargo-gccrs.git
+fi
+
+# get the log file for cargo gccrs
+pushd cargo-gccrs
+rm -f cargo.log
+gource --output-custom-log cargo.log .
+sed -i -E "s#(.+)\|#\1|/cargo#" cargo.log
+mv cargo.log ../
+popd
+
+# generate the log file for gccrs
+git log --pretty=format:user:%aN%n%ct \
+    --reverse \
+    --raw \
+    --encoding=UTF-8 \
+    --no-renames \
+    --no-show-signature \
+    -- \
+    .github \
+    Dockerfile \
+    *.md \
+    gcc/rust \
+    gcc/testsuite/rust \
+    gcc/testsuite/rust.test > gccrs-raw.log
+cat gccrs-raw.log | gource --log-format git --output-custom-log gccrs.log -
+rm -f gccrs-raw.log
+sed -i -E "s#(.+)\|#\1|/compiler#" cargo.log
+
+# combine them
+combined_log="$(mktemp /tmp/gource.XXXXXX)"
```

Comment:
Maybe remove the temp file after its use or using `trap .... EXIT` ?

Reply:
Added a cleanup step instead of a trap because sometimes killing to keep the tmp files can be useful.